File: _imported_caro_20260222_1/Controllers/ChatBotController.cs
using Microsoft.AspNetCore.Mvc;
using Microsoft.Extensions.Configuration;
using System.Net.Http;
using System.Net.Http.Headers;
using System.Text;
using System.Text.Json;
using System.Threading.Tasks;

namespace Caro.Controllers
{
    public class ChatBotController : Controller
    {
        private readonly IConfiguration _configuration;
        private readonly HttpClient _httpClient;

        public ChatBotController(IConfiguration configuration)
        {
            _configuration = configuration;
            _httpClient = new HttpClient();
        }

        public IActionResult Index()
        {
            return View();
        }

        [HttpPost]
        public async Task<IActionResult> SendMessage(string message)
        {
            var apiKey = _configuration["HuggingFace:ApiKey"];
            var model = _configuration["HuggingFace:Model"];

            _httpClient.DefaultRequestHeaders.Authorization = new AuthenticationHeaderValue("Bearer", apiKey);

            // ===== Trả lời cứng cho một số câu hỏi =====
            string normalized = message.ToLower();

            if (normalized.Contains("đấu rank") || normalized.Contains("dau rank"))
            {
                return Json(new
                {
                    success = true,
                    reply = "Hướng dẫn đấu xếp hạng:\n1. Vào mục 'Home' ở góc trên bên trái.\n2. Chọn multiplayer.\n3. Chọn 'online'.\n4. Bấm 'Tìm trận' để vào trận xếp hạng."
                });
            }

            if (normalized.Contains("xem bảng xếp hạng") || normalized.Contains("bang xep hang"))
            {
                return Json(new
                {
                    success = true,
                    reply = "Chọn 'Bảng xếp hạng' phía trên bên trái để xem thứ hạng người chơi theo điểm số."
                });
            }

            if (normalized.Contains("đổi mật khẩu") || normalized.Contains("doi mat khau"))
            {
                return Json(new
                {
                    success = true,
                    reply = "Click vào avatar góc phải trên cùng → chọn 'Đổi mật khẩu'."
                });
            }

            // ===== Gửi đến mô hình AI HuggingFace (nếu không khớp câu hỏi cứng) =====
            var requestBody = new
            {
                inputs = message
            };

            var content = new StringContent(JsonSerializer.Serialize(requestBody), Encoding.UTF8, "application/json");
            var response = await _httpClient.PostAsync($"https://api-inference.huggingface.co/models/{model}", content);

            if (!response.IsSuccessStatusCode)
            {
                var err = await response.Content.ReadAsStringAsync();
                return Json(new { success = false, message = $"Lỗi Hugging Face: {response.StatusCode} - {err}" });
            }

            var responseString = await response.Content.ReadAsStringAsync();

            try
            {
                using var doc = JsonDocument.Parse(responseString);
                var result = doc.RootElement[0].GetProperty("generated_text").GetString();
                return Json(new { success = true, reply = result });
            }
            catch
            {
                return Json(new { success = false, message = "Lỗi xử lý phản hồi từ mô hình AI!" });
            }
        }
    }
}

File: src/main/backend/java/com/game/hub/controller/ChatBotController.java
package com.game.hub.controller;

import org.springframework.beans.factory.annotation.Value;
import org.springframework.http.HttpEntity;
import org.springframework.http.HttpHeaders;
import org.springframework.http.MediaType;
import org.springframework.http.ResponseEntity;
import org.springframework.stereotype.Controller;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.ResponseBody;
import org.springframework.web.client.RestTemplate;

import java.util.List;
import java.util.Map;

@Controller
@RequestMapping("/chat-bot")
public class ChatBotController {
    private final RestTemplate restTemplate = new RestTemplate();

    @Value("${huggingface.apiKey:}")
    private String apiKey;

    @Value("${huggingface.model:}")
    private String model;

    @GetMapping
    public String page() {
        return "chat-bot/index";
    }

    @ResponseBody
    @PostMapping("/send")
    public Map<String, Object> sendMessage(@RequestBody ChatRequest request) {
        String message = request.message() == null ? "" : request.message().trim();
        String normalized = message.toLowerCase();

        if (normalized.contains("dau rank") || normalized.contains("dau xep hang")) {
            return Map.of("success", true, "reply",
                "Huong dan dau xep hang: vao Home -> Multiplayer -> Online -> Tim tran.");
        }

        if (normalized.contains("bang xep hang")) {
            return Map.of("success", true, "reply", "Mo muc Bang xep hang de xem top nguoi choi theo diem.");
        }

        if (normalized.contains("doi mat khau")) {
            return Map.of("success", true, "reply", "Vao trang Profile/Account va chon doi mat khau.");
        }

        if (apiKey == null || apiKey.isBlank() || model == null || model.isBlank()) {
            return Map.of("success", true, "reply", "Chatbot AI chua duoc cau hinh API key/model.");
        }

        try {
            String url = "https://api-inference.huggingface.co/models/" + model;
            HttpHeaders headers = new HttpHeaders();
            headers.setBearerAuth(apiKey);
            headers.setContentType(MediaType.APPLICATION_JSON);

            HttpEntity<Map<String, String>> entity = new HttpEntity<>(Map.of("inputs", message), headers);
            ResponseEntity<List> response = restTemplate.postForEntity(url, entity, List.class);

            if (!response.getStatusCode().is2xxSuccessful() || response.getBody() == null || response.getBody().isEmpty()) {
                return Map.of("success", false, "message", "Loi goi HuggingFace API");
            }

            Object first = response.getBody().get(0);
            if (first instanceof Map<?, ?> map) {
                Object generated = map.get("generated_text");
                if (generated != null) {
                    return Map.of("success", true, "reply", generated.toString());
                }
            }

            return Map.of("success", false, "message", "Khong phan tich duoc ket qua AI");
        } catch (Exception ex) {
            return Map.of("success", false, "message", "Loi chatbot: " + ex.getMessage());
        }
    }

    public record ChatRequest(String message) {
    }
}